File: Easy/HappyNumbers.java
import java.io.*;
import java.util.ArrayList;

public class Main {
    public static String solve(String input) {
        Integer number = Integer.parseInt(input);
        ArrayList<Integer> generatedNumbers = new ArrayList<>();

        while ((!isHappy(number)) && (!generatedNumbers.contains(number))) {
            generatedNumbers.add(number);
            number = getNextNumber(number);
        }

        return isHappy(number) ? "1" : "0";
    }

    private static Integer getNextNumber(Integer number) {
        Integer sum = 0;
        while (number > 0) {
            sum += (number % 10) * (number % 10);
            number /= 10;
        }
        
        return sum;
    }

    private static boolean isHappy(Integer number) {
        return number.equals(1);
    }
    
    public static void main (String[] args) throws IOException {
        File file = new File(args[0]);
        BufferedReader buffer = new BufferedReader(new FileReader(file));
        String line;
        while ((line = buffer.readLine()) != null) {
            line = line.trim();
            System.out.println(solve(line));
        }
    }
}

File: Easy/HappyNumbers.cpp
#include <iostream>
#include <fstream>
#include <sstream>
#include <set>

using namespace std;

bool isHappy(int number) {
    return number == 1;
}

int getNextNumber(int number) {
    int sum = 0;
    while (number > 0) {
        sum += (number % 10) * (number % 10);
        number /= 10;
    }
        
    return sum;
}

string solve(const string &input) {
    int number = stoi(input);
    set<int> generated_numbers;

    while ((!isHappy(number)) && (generated_numbers.find(number) == generated_numbers.end())) {
        generated_numbers.insert(number);
        number = getNextNumber(number);
    }

    return isHappy(number) ? "1" : "0";
}

int main(int argc, const char * argv[]) {
    ifstream input_stream(argv[1]);
    string input;
    
    while(getline(input_stream, input)) {
        cout << solve(input) << endl;
    }
    
    return 0;
}
